localstack/aws/protocol/service_router.py

```diff
@@ -140,10 +140,13 @@ def custom_host_addressing_rules(host: str) -> Optional[ServiceModelIdentifier]:
     """
     Rules based on the host header of the request, which is typically the data plane of a service.
 
-    # TODO: ELB, AppSync, CloudFront, ...
+    Some services are added through a patch in ext.
     """
-    if ".execute-api." in host:
-        return ServiceModelIdentifier("apigateway")
+
+    # a note on ``.execute-api.`` and why it shouldn't be added as a check here: ``.execute-api.`` was previously
+    # mapped distinctly to ``apigateway``, but this assumption is too strong, since the URL can be apigw v1, v2,
+    # or apigw management api. so in short, simply based on the host header, it's not possibly to unambiguously
```

Comment:
nit: Small typo, which made me take embarrassingly long to read this sentence 😛 (please don't add a co-authorship for this though 😅).
```suggestion
    # or apigw management api. so in short, simply based on the host header, it's not possible to unambiguously
```